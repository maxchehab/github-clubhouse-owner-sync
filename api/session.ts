import { NextApiResponse, NextApiRequest } from 'next';
import withSession from '../common/util/with-session.util';

import lambda from '../common/util/lambda.util';
import Session from '../common/interfaces/session.interface';

export default lambda(
  withSession(
    async (_req: NextApiRequest, res: NextApiResponse, session: Session) => {
      delete session.token;

      return res.json(session);
    },
  ),
);
