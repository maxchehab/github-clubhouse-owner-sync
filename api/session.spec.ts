import axios from 'axios';
import http from 'http';
import listen from 'test-listen';

import * as SessionService from '../services/fauna/services/session.service';
import closeServer from '../test/close-server';
import createTestLambda from '../test/create-test-lambda';
import session from './session';

describe('api/github/session', () => {
  let server: http.Server;
  let url: string;

  beforeAll(async () => {
    server = createTestLambda(session);
    url = await listen(server);
  });

  afterAll(async () => {
    await closeServer(server);
  });

  describe('with GET api/github/session', () => {
    beforeEach(() => {
      jest
        .spyOn(SessionService, 'findSessionByID')
        .mockImplementationOnce(async () => ({
          name: 'John Smith',
          expires_at: new Date().getTime() + 1000,
          token: 'access-token',
          id: 'sess_01DY49R4RBSET4H39RX7JKR0AG',
          avatar: 'https://example.com/avatar.png',
        }));
    });

    it('returns session without token', async () => {
      const { status, data } = await axios.get(url, {
        headers: {
          cookie: 'sid=sess_01DY49R4RBSET4H39RX7JKR0AG',
        },
      });

      expect(status).toEqual(200);
      expect(data).toMatchSnapshot({ expires_at: expect.any(Number) });
    });
  });
});
