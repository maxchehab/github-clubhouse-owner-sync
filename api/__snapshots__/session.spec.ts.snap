// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/github/session with GET api/github/session returns session without token 1`] = `
Object {
  "expires_at": 1578548737237,
  "id": "sess_01DY49R4RBSET4H39RX7JKR0AG",
  "name": "John Smith",
}
`;
